docker/docker_jit_monitor/src/main.rs

```diff
@@ -219,34 +297,45 @@ fn main() -> anyhow::Result<()> {
     let mut running_hos_runners = vec![];
     // Todo: implement something to reserve devices for the duration of the docker run child process.
     const MAX_HOS_RUNNERS: usize = 1;
+    let mut retries = 0;
```

Comment:
We probably want to seperate this for the builder and runner. If e.g. the runner image is broken, but the builder image is fine, then the builder spawning would reset the runner retry count, which we wouldn't want.